.github/workflows/CI.yml

```diff
@@ -38,82 +38,38 @@ jobs:
     name: cargo check
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v3
-    - uses: actions-rs/toolchain@v1
-      with:
-        toolchain: stable
-        profile: minimal
-        override: true
-    - name: Check
-      uses: actions-rs/cargo@v1
-      with:
-        command: check
-        args: --all --tests --benches
+      - uses: actions/checkout@v3
+      - uses: dtolnay/rust-toolchain@stable
+      - name: Check
+        run: cargo check --all --tests --benches
 
   style:
     # Check style.
     name: cargo fmt
     needs: check
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v3
-    - uses: actions-rs/toolchain@v1
-      with:
-        toolchain: stable
-        components: rustfmt
-        profile: minimal
-        override: true
-    - name: rustfmt
-      uses: actions-rs/cargo@v1
-      with:
-        command: fmt
-        args: --all -- --check
+      - uses: actions/checkout@v3
+      - uses: dtolnay/rust-toolchain@stable
+        with:
+          components: rustfmt
+      - name: rustfmt
+        run: cargo fmt --all -- --check
 
   warnings:
     # Check for any warnings. This is informational and thus is allowed to fail.
     runs-on: ubuntu-latest
     needs: check
-    steps:
-    - uses: actions/checkout@v3
-    - uses: actions-rs/toolchain@v1
-      with:
-        toolchain: stable
-        components: clippy
-        profile: minimal
-    - name: Clippy
-      uses: actions-rs/clippy-check@v1
-      with:
-        token: ${{ secrets.GITHUB_TOKEN }}
-        args: --all --examples --tests --benches -- -D warnings
-
-  minimal-versions:
-    # Check for minimal-versions errors where a dependency is too
-    # underconstrained to build on the minimal supported version of all
-    # dependencies in the dependency graph.
-    name: cargo check (-Zminimal-versions)
-    needs: check
-    runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v3
-      - uses: actions-rs/toolchain@v1
+      - uses: dtolnay/rust-toolchain@stable
         with:
-          toolchain: nightly
-          profile: minimal
-          override: true
-      - name: install cargo-hack
-        uses: taiki-e/install-action@cargo-hack
-      - name: "check --all-features -Z minimal-versions"
-        run: |
-          # Remove dev-dependencies from Cargo.toml to prevent the next `cargo update`
-          # from determining minimal versions based on dev-dependencies.
-          cargo hack --remove-dev-deps --workspace
-          # Update Cargo.lock to minimal version dependencies.
-          cargo update -Z minimal-versions
-          cargo hack check \
-            --package tracing \
-            --package tracing-core \
-            --package tracing-subscriber \
-            --all-features --ignore-private
```

Comment:
Yes, the theory is that checking `+MSRV -Zminimal-versions` is sufficient, and that standard backwards-compatibility guarantees imply that `+stable -Zminimal-versions` will also work. This isn't *guaranteed*, but we can have fair confidence since we're testing stable with normal maximal version resolution.